### PR TITLE
feat: [247] dont show 3.4 nested question for 'unknown' and 'none' op…

### LINKED ISSUE #247: [BUG] Shouldn't have to define which are primary and co-benefit for "none" and "unknown"
image

Question 3.1, 3.2, 3.3
When you select "none" and/or "unknown", it still prompts you to select how stakeholders benefit. This doesn't make sense for none and unknow options since there are no aims for the stakeholders to benefit from.

### PR DESCRIPTION
**Pull request by mmnoo**
…tions for section 3 

Fixes #247 

Steps for testing:
- log in
- create a new site, fill out a couple of stakeholders (section 2)
- go do section 3/restoration aims
- select an aim. See that the stakeholder benefits section shows for all options, but not for 'Unknown' or 'None'

### CODE REVIEW

**Comment by julia-rom** [comment]
I just noticed the second section is breaking @mmnoo , can't access the page without an error 
Screen Shot 2022-08-29 at 10 51 07 AM

**Review by julia-rom — CHANGES_REQUESTED** [review]
breaking change - see screenshot

**Comment by mmnoo** [comment]
> I just noticed the second section is breaking @mmnoo , can't access the page without an error Screen Shot 2022-08-29 at 10 51 07 AM

Ahhhhhh! thanks!

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js [hunk 1]
```diff
@@ -180,6 +187,7 @@ CheckboxGroup.propTypes = {
 
 CheckboxGroup.defaultProps = {
   onBlur: () => {},
+  optionsExcludedFromShowingSelectedMarkup: [],
```
Comment (mmnoo) [inline mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:190]: This is really weird, the default prop isnt working and Im puzzled as to why. The composing component isnt setting this prop, so it should be an empty array....

File: mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js [hunk 2]
```diff
@@ -28,13 +28,14 @@ const CheckboxGroup = forwardRef(
...
       options,
-      shouldAddOtherOptionWithClarification,
-      value,
-      SelectedMarkup,
-      shouldReturnEvent
+      optionsExcludedFromShowingSelectedMarkup = [],
+      shouldAddOtherOptionWithClarification = false,
+      value = [],
+      SelectedMarkup = undefined,
+      shouldReturnEvent = false
```
Comment (mmnoo) [inline mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:38]: react `defaultProps` wasnt working and apparently is slotted for depreciation, so I cant help but wonder if they arent bothering to fix its bugs. This fixed the bug you found.

Anyway, I'm leaning toward using the es6 way. Curious to hear the opinions of others in the web guild thread.

Comment (julia-rom) [reply mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js:38]: interesting! also nice find. Will note not to use defaultProps from now on (or just use with caution). Going to read more into it

**Review by julia-rom — APPROVED** [review]

**Pull request merged**